### PR TITLE
Add quickstart

### PR DESCRIPTION
**Pull request by jerpint**

### CODE REVIEW

File: buster/docparser.py [hunk 1]
```diff
@@ -84,25 +89,89 @@ def get_all_documents(
...
-    encoding = tiktoken.get_encoding(EMBEDDING_ENCODING)
+def compute_n_tokens(
+    df: pd.DataFrame, embedding_encoding: str = EMBEDDING_ENCODING, col: str = "content"
+) -> pd.DataFrame:
+    """Counts the tokens in the content column and adds the count to a n_tokens column."""
+    logger.info("Computing tokens counts...")
+    encoding = tiktoken.get_encoding(encoding_name=embedding_encoding)
     # TODO are there unexpected consequences of allowing endoftext?
-    df["n_tokens"] = df.content.apply(lambda x: len(encoding.encode(x, allowed_special={"<|endoftext|>"})))
+    df[f"n_tokens"] = df[col].apply(lambda x: len(encoding.encode(x, allowed_special={"<|endoftext|>"})))
```
Comment (hbertrand) [inline buster/docparser.py:99]: why make it a f-string?

File: buster/docparser.py [hunk 2]
```diff
@@ -84,25 +89,89 @@ def get_all_documents(
...
+
+    trimmed = df[word_counts_before == word_counts_after]
+    logger.info(f"trimmed {len(trimmed)} documents to {max_words} words.")
+
     return df
 
 
-def generate_embeddings(root_dir: str, output_filepath: str, source: str) -> pd.DataFrame:
-    # Get all documents and precompute their embeddings
+def compute_embeddings(df: pd.DataFrame, engine: str = EMBEDDING_MODEL) -> pd.DataFrame:
```
Comment (hbertrand) [inline buster/docparser.py:116]: maybe add a `col` argument to be consistent with the other changes?

File: buster/docparser.py [hunk 3]
```diff
@@ -84,25 +89,89 @@ def get_all_documents(
...
+def generate_embeddings_parser(root_dir: str, output_filepath: str, source: str) -> pd.DataFrame:
     documents = get_all_documents(root_dir, supported_docs[source]["base_url"], supported_docs[source]["parser"])
+    return generate_embeddings(documents, output_filepath)
+
+
+def documents_to_db(documents: pd.DataFrame, output_filepath: str):
+    logger.info("Preparing database...")
+    sources = documents["source"].unique()
+    for source in sources:
+        documents_manager = get_documents_manager_from_extension(output_filepath)(output_filepath)
```
Comment (hbertrand) [inline buster/docparser.py:132]: put this line outside the loop

File: buster/docparser.py [hunk 4]
```diff
@@ -84,25 +89,89 @@ def get_all_documents(
...
 
-    documents_manager = get_documents_manager_from_extension(output_filepath)(output_filepath)
-    documents_manager.add(source, documents)
+    # save the documents to a db for later use
+    documents_to_db(documents, output_filepath)
 
     return documents
+
+
+@click.command()
```
Comment (hbertrand) [inline buster/docparser.py:160]: didn't know about click, it looks nice

**Review by hbertrand — CHANGES_REQUESTED** [review]

File: README.md [hunk 5]
```diff
@@ -11,12 +11,38 @@ pinned: false
...
 
-![Question: How do I load a Huggingface model?](buster/imgs/qa_web_load.png)
+You can try out our [live demo here](https://huggingface.co/spaces/jerpint/buster), where it will answer questions about a bunch of libraries we've already scraped, including [🤗 Transformers](https://huggingface.co/docs/transformers/index).
+
+
+# Quickstart
+
+Here is a quick guide to help you deploy buster on your own dataset!
+
+First step, install buster locally. Note that buster requires python>=3.10.
```
[the remark this answers is not in the document]
Comment (jerpint) [reply README.md:25]: agreed

[the remark this answers is not in the document]
Comment (jerpint) [reply README.md:25]: opened an issue for that in #75

File: buster/docparser.py (hunk 1 above)
Comment (jerpint) [reply buster/docparser.py:99]: no idea, must have slipped in, removed

File: buster/docparser.py [hunk 6]
```diff
@@ -84,25 +89,89 @@ def get_all_documents(
...
-    df["n_tokens"] = df.content.apply(lambda x: len(encoding.encode(x, allowed_special={"<|endoftext|>"})))
+    df[f"n_tokens"] = df[col].apply(lambda x: len(encoding.encode(x, allowed_special={"<|endoftext|>"})))
     return df
 
 
-def precompute_embeddings(df: pd.DataFrame) -> pd.DataFrame:
-    df["embedding"] = df.content.apply(lambda x: np.asarray(get_embedding(x, engine=EMBEDDING_MODEL), dtype=np.float32))
+def max_word_count(df: pd.DataFrame, max_words: int, col: str = "content") -> pd.DataFrame:
+    """Trim the word count of an entry to max_words"""
+    assert df[col].apply(lambda s: isinstance(s, str)).all(), "Column {col} must contain only strings"
```
[the remark this answers is not in the document]
Comment (jerpint) [reply buster/docparser.py:105]: good catch

File: buster/docparser.py [hunk 7]
```diff
@@ -84,25 +89,89 @@ def get_all_documents(
...
+    word_counts_before = df[col].apply(lambda x: len(x.split(" ")))
+    df[col] = df[col].apply(lambda x: " ".join(x.split()[:max_words]))
+    word_counts_after = df[col].apply(lambda x: len(x.split(" ")))
```
[the remark this answers is not in the document]
Comment (jerpint) [reply buster/docparser.py:108]: hehe oops, fixed

File: buster/docparser.py [hunk 8]
```diff
@@ -84,25 +89,89 @@ def get_all_documents(
...
+    documents: pd.DataFrame,
+    output_filepath: str = "documents.db",
+    max_words=500,
+    embedding_engine: str = EMBEDDING_MODEL,
+) -> pd.DataFrame:
+    # check that we have the appropriate columns in our dataframe
+
+    assert set(required_cols := ["content", "title", "url"]).issubset(
+        set(documents.columns)
+    ), f"Your datafraem must contain {required_cols}."
```
[the remark this answers is not in the document]
Comment (jerpint) [reply buster/docparser.py:147]: fixed

File: buster/docparser.py (hunk 4 above)
Comment (jerpint) [reply buster/docparser.py:160]: yes, alternative to argparse, simplifies the syntax or cli args

**Review by hbertrand — APPROVED** [review]

**Pull request merged**